master/internal/db/postgres_trial.go

```diff
@@ -212,21 +272,16 @@ func (db *PgDB) AddValidationMetrics(
 			return err
 		}
 
-		if _, err := tx.ExecContext(ctx, `
+		// QUESTION: why don't we check this with raw_steps as well same as AddTrainingMetrics?
```

Comment:
need to remove once answered. 
I think it might be because we want to do validations after a certain amount of training. So we remove validation if the training metrics are removed. But not vice versa. @hamidzr @ioga

Reply:
ah I think this makes the logic between the two `add*Metric` very similar with just the target table name being different. I'll see about having these just call the same implementation